include/libtorrent/upnp.hpp

```diff
@@ -120,9 +120,30 @@ struct parse_state
 	}
 };
 
+struct error_code_parse_state
+{
+	error_code_parse_state(): in_error_code(false), exit(false), error_code(-1) {}
+	bool in_error_code;
+	bool exit;
+	int error_code;
+};
+
+struct ip_address_parse_state: public error_code_parse_state
```

Comment:
the ``public`` here is redundant. for ``struct``, the default inheritance is default.